### PR TITLE
LibWeb: Align NavigationParams and the creation AOs to the spec 

### PR DESCRIPTION
**Pull request by ADKaster**
I started this branch trying to add the NavigateEvent AOs to fire the event so I could write tests, but ended up doing all this cleanup instead :upside_down_face:

### CODE REVIEW

File: Userland/Libraries/LibWeb/HTML/Navigable.cpp [hunk 1]
```diff
@@ -379,8 +381,86 @@ Vector<JS::NonnullGCPtr<SessionHistoryEntry>>& Navigable::get_session_history_en
...
+
+    // 7. Return policy.
+    return policy;
+}
+
+// https://html.spec.whatwg.org/multipage/browsing-the-web.html#attempt-to-create-a-non-fetch-scheme-document
+static JS::GCPtr<DOM::Document> attempt_to_create_a_non_fetch_scheme_document(NonFetchSchemeNavigationParams const&)
+{
+    // FIXME: Implement this algorithm to hand off to external software or display inline content
+    return nullptr;
```
Comment (kalenikaliaksandr) [inline Userland/Libraries/LibWeb/HTML/Navigable.cpp:459]: I would like to see FIXME message saying that we can't create document for the url if we reached this function.

Comment (ADKaster) [reply Userland/Libraries/LibWeb/HTML/Navigable.cpp]: ah good point. now we print out that we don't know how to navigate to that URL to dbgln

File: Userland/Libraries/LibWeb/DOM/Document.h [hunk 2]
```diff
@@ -445,6 +445,10 @@ class Document
     bool is_initial_about_blank() const { return m_is_initial_about_blank; }
     void set_is_initial_about_blank(bool b) { m_is_initial_about_blank = b; }
 
+    // https://html.spec.whatwg.org/multipage/dom.html#concept-document-about-base-url
+    Optional<AK::URL> about_base_url() const { return m_about_base_url; }
```
Comment (kalenikaliaksandr) [inline Userland/Libraries/LibWeb/DOM/Document.h:449]: cool, now we can pass `about base url` to document state here:
https://html.spec.whatwg.org/multipage/document-sequences.html#child-navigables:concept-document-about-base-url
and here
https://html.spec.whatwg.org/multipage/document-sequences.html#top-level-traversables:concept-document-about-base-url
and probably there are more places.
you want to fix that in this PR? :)

Comment (ADKaster) [reply Userland/Libraries/LibWeb/DOM/Document.h:449]: Let's try not to add any more uses of the pattern of only including half of the spec comment :sweat_smile: 

Not sure how I would have found those easily without you mentioning it lol

Comment (ADKaster) [reply Userland/Libraries/LibWeb/DOM/Document.h:449]: Unless the spec changed under us to add this concept, in which case: good catch!

Comment (ADKaster) [reply Userland/Libraries/LibWeb/DOM/Document.h:449]: oh jeez. create_a_new_browsing_context_and_document needs some help to look like the current spec steps. But, I found the spec cross-references pretty useful for finding all the uses :+1:

**Pull request merged**